src/controllers/user/authenticate.js

```diff
@@ -15,14 +15,44 @@ const allowlist = [
 
 const adapter = new RedisAdapter('AccessToken')
 
-const checkForAuthentication = async (ctx, next) => {
+const setAccessTokenOnContext = async (ctx) => {
   if (ctx.get('Authorization').startsWith('Bearer ')) {
     // bearer auth
     ctx.accessToken = ctx.get('Authorization').slice(7, ctx.get('Authorization').length).trimLeft()
   } else if (ctx.session.grant && ctx.session.grant.response) {
     // session auth
     ctx.accessToken = ctx.session.grant.response.access_token
   }
+}
+
+const findSession = async (ctx) => {
+  if (!ctx.accessToken) {
+    return null
+  }
+  const session = await adapter.find(ctx.accessToken)
+  if (session.accountId) {
+    const user = await User.findOne({
+      where: {
+        id: session.accountId
+      },
+      raw: true
+    })
+
+    return user
+  }
+  return null
+}
+
+module.exports.loadProfileIntoContext = async (ctx, next) => {
```

Comment:
Useful for if a user is authenticated and we want to change the response based on that.